lib/AST/LayoutConstraint.cpp

```diff
@@ -106,6 +110,24 @@ bool LayoutConstraintInfo::isNativeRefCountedObject(LayoutConstraintKind Kind) {
   return Kind == LayoutConstraintKind::NativeRefCountedObject;
 }
 
+bool LayoutConstraintInfo::isClass(LayoutConstraintKind Kind) {
+  return Kind == LayoutConstraintKind::Class;
+}
+
+bool LayoutConstraintInfo::isNativeClass(LayoutConstraintKind Kind) {
+  return Kind == LayoutConstraintKind::NativeClass;
+}
+
+bool LayoutConstraintInfo::isRefCounted(LayoutConstraintKind Kind) {
+  return Kind == LayoutConstraintKind::RefCountedObject ||
+         Kind == LayoutConstraintKind::Class;
```

Comment:
Shouldn't this also check for `Kind == LayoutConstraintKind::NativeRefCountedObject` and `Kind == LayoutConstraintKind::NativeClass`?